### PR TITLE
Bug fix: group count

### PR DESCRIPTION
**Pull request by sashankaryal**
## What changes are proposed in this pull request?

- Fix bug where group count wasn't showing in `group` statistics aggregation mode
- Add E2E test
- Add other useful POMs

## How is this patch tested? If it is not, please explain why.

E2E test

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

**Review by benjaminpkane — APPROVED** [review]
LGTM. Can the base be changed to `release/v0.21.5`?

For confirmation, we are keeping the `data-cy` for historical reasons?

File: e2e-pw/src/oss/poms/action-row/display-options.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async setSidebarStatisticsMode(mode: SidebarStatisticsMode) {
+    const selector = this.page.getByTestId(
+      `tab-option-View ${mode} sidebar statistics`
+    );
+    return selector.click();
```
Comment (manivoxel51) [inline e2e-pw/src/oss/poms/action-row/display-options.ts:20]: do we need `await` here? 🤔

File: e2e-pw/src/oss/poms/grid/index.ts [hunk 2]
```diff
@@ -29,4 +37,17 @@ class GridAsserter {
     const lookersCount = await this.gridPom.grid.getByTestId("looker").count();
     expect(lookersCount).toBe(n);
   }
+
+  async waitForEntryCountTextToEqual(text: string) {
+    return this.gridPom.page.waitForFunction(
```
Comment (manivoxel51) [inline e2e-pw/src/oss/poms/grid/index.ts:42]: cool! I'll see if I can concert those waitForTimeouts

**Comment by sashankaryal** [comment]
> LGTM. Can the base be changed to `release/v0.21.5`?
> 
> For confirmation, we are keeping the `data-cy` for historical reasons?

Once we're completely out of the cypress territory, we can change it to `data-testid`, the default for Playwright.

**Review by manivoxel51 — COMMENTED** [review]
nice work! 🍨 

we should talk about how to choose good `testId` attributes  best practices in tech time maybe

File: e2e-pw/src/oss/poms/action-row/display-options.ts (hunk 1 above)
Comment (sashankaryal) [reply e2e-pw/src/oss/poms/action-row/display-options.ts:20]: We could `await` or just return the promise and expect the caller to await.

**Pull request merged**